packages/i18next/src/lib/InternationalizationHandler.ts

```diff
@@ -214,4 +216,16 @@ export class InternationalizationHandler {
 
 		return { namespaces: [...new Set(namespaces)], languages };
 	}
+
+	public async reloadResources() {
+		const result = await fromAsync(async () => {
+			const { namespaces, languages } = await this.walkLanguageDirectory(this.languagesDirectory);
+			await i18next.reloadResources(this.options.hmr?.languages ?? languages, this.options.hmr?.namespaces ?? namespaces);
```

Comment:
```suggestion
			let languages = this.options.hmr?.languages;
			let namespaces = this.options.hmr?.namespaces;

			if (!languagesFromOptions || !namespacesFromOptions) {
				const languageDirectoryResult = await this.walkLanguageDirectory(this.languagesDirectory);

				languages ??= languageDirectoryResult.languages;
				namespaces ??= languageDirectoryResult.namespaces;
			}

			await i18next.reloadResources(languages, namespaces);
```

We live in the future where [Logical Nullish Assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_nullish_assignment) can power up code a lot ;)

This way the `walkLanguageDirectory` won't be executed if both fields are provided through options, and only those that were not in options will be overridden by the result from `walkLanguageDirectory`.